client/src/components/feedback/edit-question-for-queue-modal.tsx

```diff
@@ -0,0 +1,271 @@
+            data-cy="create-question-modal"
+            aria-describedby="transition-modal-description"
+            className={classes.modal}
+            open={open}
+            onClose={okButtonClicked}
+            closeAfterTransition
+            BackdropComponent={Backdrop}
+            BackdropProps={{
+              timeout: 500,
+            }}
+          >
+            <Fade in={open}>
+              <div className={classes.paper}>
+                <Grid item alignItems="center" xs={12} md={12}>
+                  <TextField
+                    // user-asked question (editable)
+                    label="Create new question:"
+                    required={true}
+                    value={customQuestion} // fill in with OG user question
+                    // create a new question for mentor with (un)edited string
+                    onChange={(e) =>
+                      onTextInputChanged(e, () => {
+                        setCustomQuestion(e.target.value);
+                      })
+                    }
+                    className={classes.inputField}
+                  />
+                  <div
+                    className={classes.inputField}
+                    style={{ textAlign: "left" }}
+                  >
+                    <FormControl>
+                      <InputLabel>Subject</InputLabel>
+                      <Select
+                        // a dropdown for the subject
+                        required={true}
+                        data-cy="subject-drop-down"
+                        style={{ width: 200 }}
+                        onChange={(
+                          event: React.ChangeEvent<{ value: unknown }>
+                        ) => {
+                          setSelectedSubject(
+                            mentor.subjects.find((a) => {
+                              return a._id === event.target.value;
+                            })
+                          );
+                        }}
+                      >
+                        <option value={undefined} selected disabled>
+                          --Select a subject--
+                        </option>
+                        {mentor.subjects.map((subject) => (
+                          <option
+                            data-cy={`Subject-option-${subject._id}`}
+                            key={subject.name}
+                            value={subject._id}
+                          >
+                            {subject.name}
+                          </option>
+                        ))}
+                      </Select>
+                    </FormControl>
+                    <FormControl>
+                      <InputLabel>Category</InputLabel>
+                      <Select
+                        // a dropdown for the category
+                        data-cy="category-drop-down"
+                        value={selectedSubject?.categories} // all the subject's categories
```

Comment:
It's probably better to just not render the category and topic dropdowns until a subject is selected. What's the behavior of this dropdown if a subject isn't selected yet?